### PR TITLE
Problem: converted native logs don't have contract address

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- do the conversion at execution site.

Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/statedb/statedb.go [hunk 1]
```diff
@@ -526,3 +535,22 @@ func (s *StateDB) Commit() error {
...
+		return
+	}
+
+	if len(events) == 0 {
+		return
+	}
+
+	for _, event := range events {
+		if converter, ok := s.eventConverters[event.Type]; ok {
+			for _, log := range converter(event.Attributes) {
```
Comment (mmsqe) [inline x/evm/statedb/statedb.go:550]: Do we need convert all attributes?

Comment (yihuang) [reply x/evm/statedb/statedb.go:550]: leave the converter to decide

**Review by mmsqe — APPROVED** [review]

File: x/evm/statedb/statedb.go [hunk 2]
```diff
@@ -526,3 +535,22 @@ func (s *StateDB) Commit() error {
...
+				log.Address = contract
```
Comment (mmsqe) [inline x/evm/statedb/statedb.go:551]: Do we update BlockNumber here as well?

Comment (yihuang) [reply x/evm/statedb/statedb.go:551]: I'm doing another PR to handle block number and block hash.

File: x/evm/statedb/interfaces.go [hunk 3]
```diff
@@ -25,6 +25,8 @@ import (
 type Keeper interface {
 	// for cache store wrapping
 	StoreKeys() map[string]storetypes.StoreKey
+	// for precompile native event convertion
```
Comment (yihuang) [inline x/evm/statedb/interfaces.go]:
```suggestion
	// for precompile native event conversion
```

**Pull request merged**